src/core/model/eraser/ErasableStroke.cpp

```diff
@@ -4,300 +4,386 @@
 }
 
-auto ErasableStroke::erasePart(double x, double y, double halfEraserSize, PartList::iterator& partIter, PartList& list,
-                               bool* deleteStrokeAfter) -> bool {
-    bool changed = false;
-
-    partIter->splitFor(halfEraserSize);
+std::vector<ErasableStroke::SubSection> ErasableStroke::getRemainingSubSectionsVector() const {
+    std::lock_guard<std::mutex> lock(this->sectionsMutex);
+    return this->remainingSections.cloneToIntervalVector();
+}
 
-    double x1 = x - halfEraserSize;
-    double x2 = x + halfEraserSize;
-    double y1 = y - halfEraserSize;
-    double y2 = y + halfEraserSize;
+bool ErasableStroke::isClosedStroke() const { return this->closedStroke; }
 
+Rectangle<double> ErasableStroke::getSubSectionBoundingBox(const ErasableStroke::SubSection& section) const {
     /**
-     * erase the beginning
+     * First look for the box in the cache
      */
-
-    std::vector<Point>& points = partIter->getPoints();
-
-    for (auto pointIter = points.begin(); pointIter != points.end();) {
-        if (pointIter->x >= x1 && pointIter->y >= y1 && pointIter->x <= x2 && pointIter->y <= y2) {
-            pointIter = points.erase(pointIter);
-            changed = true;
-        } else {
-            // only the beginning is handled here
-            break;
-        }
+    auto it = std::lower_bound(boundingBoxes.begin(), boundingBoxes.end(), section,
+                               [](const std::pair<SubSection, Rectangle<double>>& cacheData,
+                                  const SubSection& section) { return cacheData.first < section; });
+    if (it != boundingBoxes.end() && section == it->first) {
+        // There was already a box computed for this section
+        return it->second;
     }
 
-    /**
-     * erase the end
-     */
-    // ugly loop avoiding reverse_iterators
-    for (auto pointIter = points.end() - 1; pointIter-- != points.begin();) {
-        if (pointIter->x >= x1 && pointIter->y >= y1 && pointIter->x <= x2 && pointIter->y <= y2) {
-            points.erase(pointIter);
-            changed = true;
-        } else {
-            // only the end is handled here
-            break;
-        }
+    // Need to compute the bounding box
+    Point p = this->stroke.getPoint(section.min);
+    double minX = p.x;
+    double maxX = p.x;
+    double minY = p.y;
+    double maxY = p.y;
+
+    auto data = this->stroke.getPointVector();
+    auto endIt = std::next(data.cbegin(), (std::ptrdiff_t)section.max.index + 1);
+    for (auto ptIt = std::next(data.cbegin(), (std::ptrdiff_t)section.min.index + 1); ptIt != endIt; ++ptIt) {
+        minX = std::min(minX, ptIt->x);
+        maxX = std::max(maxX, ptIt->x);
+        minY = std::min(minY, ptIt->y);
+        maxY = std::max(maxY, ptIt->y);
     }
 
+    Point q = this->stroke.getPoint(section.max);
+    minX = std::min(minX, q.x);
+    maxX = std::max(maxX, q.x);
+    minY = std::min(minY, q.y);
+    maxY = std::max(maxY, q.y);
```

Comment:
Isn't the point already computed in the for loop?

Reply:
See above.